store/store.go

```diff
@@ -59,11 +64,17 @@ func newStoreWithRetry(path string, maxRetries int) (kv.Storage, error) {
 	}
 
 	name := strings.ToLower(storeURL.Scheme)
+
+	storesLock.RLock()
+
 	d, ok := stores[name]
```

Comment:
Move two `storesLock.RUnlock()` right after this line?

Reply:
... or write a function to retrieve the driver, like:

```go
func loadDriver(name string) (kv.Driver, bool) {
	storesLock.RUnlock()
	defer storesLock.Unlock()
	d, ok := stores[name]
	return d, ok
}
```

Reply:
I'd prefer write a function to get the driver , and scope for lock is more less in this way